### PR TITLE
Doc updates

### PR DESCRIPTION
**Pull request by jsmartt**
- Updates README with more complete information about properties (if they are optional, what the default is, etc.). Also fixes a few inconsistencies in it.
 - Fixes/updates the properties in each resource to match the README. This includes setting defaults, requiring certain params for certain actions, etc.
 - Updates test recipes to use common attributes for ilo information (and adds license text)

### CODE REVIEW

**Comment by jsmartt** [comment]
Hmm. Looks like the Ruby 2.1 build couldn't install [buff-extensions 2.0](https://rubygems.org/gems/buff-extensions/versions/2.0.0); it requires Ruby >= 2.2.0.  Not sure exactly how we can fix this without removing getting rid of the tests on 2.1. I don't want to have to set that dependency in the Gemfile, because it's not a direct dependency; looks like berkshelf is the one requiring it.

While we're at it, we might as well add in 2.3 also.

Thoughts @Anirudh-Gupta or @vivekbhatia14 ?

File: libraries/log_entry.rb [hunk 1]
```diff
@@ -17,14 +17,15 @@ class LogEntry < BaseResource
...
-    property :log_type, String, required: true, equal_to: ['IEL', 'IML']
+    property :log_type, String, required: true, equal_to: ['IEL', 'IML', :IEL, :IML]
     property :severity_level, String, equal_to: ['OK', 'Warning', 'Critical']
     property :dump_file, String
     property :owner, [String, Integer], default: ENV['USER'] || ENV['USERNAME']
     property :group, [String, Integer], default: ENV['USER'] || ENV['USERNAME']
     property :duration, Integer, default: 24
 
     action :dump do
+      raise 'Please set the :dump_file property (String)' unless dump_file
```
Comment (Anirudh-Gupta) [inline libraries/log_entry.rb]: why not??  ``` property :dump_file, String, required: true``` Why ain't we leveraging Chef's inbuilt validations if we don't need to put in custom message?

Comment (jsmartt) [inline libraries/log_entry.rb]: We definitely don't need to, but it just makes the error message a little bit more clear. With it gone, the error is:

```
    ================================================================================
    Error executing action `dump` on resource 'ilo_log_entry[set a subset of BIOS settings]'
    ================================================================================

    ArgumentError
    -------------
    You must supply a name when declaring a file resource

    Cookbook Trace:
    ---------------
    /home/vagrant/chef-repo/.chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/copied_from_chef/chef/resource_builder.rb:63:in `build'
    /home/vagrant/chef-repo/.chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/copied_from_chef/chef/dsl/declare_resource.rb:302:in `build_resource'
    /home/vagrant/chef-repo/.chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/copied_from_chef/chef/dsl/declare_resource.rb:259:in `declare_resource'
    /home/vagrant/chef-repo/.chef/local-mode-cache/cache/cookbooks/ilo/libraries/log_entry.rb:36:in `block in <class:LogEntry>'
    /home/vagrant/chef-repo/.chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/copied_from_chef/chef/provider.rb:132:in `instance_eval'
    /home/vagrant/chef-repo/.chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/copied_from_chef/chef/provider.rb:132:in `compile_and_converge_action'
    /home/vagrant/chef-repo/.chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:78:in `run_action'
    /home/vagrant/chef-repo/.chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `block (2 levels) in converge'
    /home/vagrant/chef-repo/.chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `each'
    /home/vagrant/chef-repo/.chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `block in converge'
    /home/vagrant/chef-repo/.chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:105:in `converge'

    Resource Declaration:
    ---------------------
    # In /home/vagrant/chef-repo/.chef/local-mode-cache/cache/cookbooks/ilo/recipes/log_test.rb

      6: ilo_log_entry 'set a subset of BIOS settings' do
      7:   ilos my_ilos
      8:   log_type 'IEL'
      9: end

    Compiled Resource:
    ------------------
    # Declared in /home/vagrant/chef-repo/.chef/local-mode-cache/cache/cookbooks/ilo/recipes/log_test.rb:6:in `from_file'

    ilo_log_entry("dump IEL logs") do
      action [:dump]
      retries 0
      retry_delay 2
      default_guard_interpreter :default
      declared_type :ilo_log_entry
      cookbook_name "ilo"
      recipe_name "log_test"
      log_type "IEL"
    end

    Platform:
    ---------
    x86_64-linux

[2016-09-07T19:21:35+00:00] INFO: Running queued delayed notifications before re-raising exception

Running handlers:
[2016-09-07T19:21:35+00:00] ERROR: Running exception handlers
Running handlers complete
[2016-09-07T19:21:35+00:00] ERROR: Exception handlers complete
Chef Client failed. 0 resources updated in 06 seconds
[2016-09-07T19:21:35+00:00] FATAL: Stacktrace dumped to /home/vagrant/chef-repo/.chef/local-mode-cache/cache/chef-stacktrace.out
[2016-09-07T19:21:35+00:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
[2016-09-07T19:21:35+00:00] ERROR: ilo_log_entry[set a subset of BIOS settings] (ilo::log_test line 6) had an error: ArgumentError: You must supply a name when declaring a file resource
[2016-09-07T19:21:35+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)

```

So it's clear which resource it's coming from, but maybe not clear exactly what you need to do. It's not necessary, but I don't think it hurts.

Comment (Anirudh-Gupta) [inline libraries/log_entry.rb]: Makes sense.

**Pull request merged**